### PR TITLE
fix: nav menu accessibility

### PR DESCRIPTION
**Pull request by bbsmooth**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

If this is too much for one PR let me know and I will split it up. 

- Move menu toggle button before list of links
- Add necessary aria attributes to menu toggle button
- Add sr only text announcing that menu links open in new window
- Add external link icon to menu links for visual cue that links open in new window
- Add keyboard focus outline around all links and button in nav (it was necessary to update some of the CSS for the nav menu in order to get all four sides of the focus outline to show)
- Use the :focus-visible pseudo class to only show focus outline when keyboard is used
- Change URL of freeCodeCamp link to main home page. It was pointing to the coderadio page which doesn't make sense and is bad for accessibility. If you don't want this to link to the home page then I suggest we don't make it a link.

### CODE REVIEW

File: src/components/Nav.js [hunk 1]
```diff
@@ -33,23 +46,17 @@ export default function Nav() {
             {links.map((link, index) => (
               <li key={index}>
                 <a href={link.href} rel='noopener noreferrer' target='_blank'>
-                  {link.text}
+                  <span>
+                    {link.text}{' '}
+                    <span className='sr-only'>opens in new window</span>
```
Comment (ahmadabdolsaheb) [inline src/components/Nav.js:51]: So we would not need to display these icons when the navigation items are not collapsed.
I understand the confusion as the usage is not documented yet.

Comment (bbsmooth) [reply src/components/Nav.js:51]: Sorry, I'm not sure I understand this comment. When the menu is collapsed then the `<ul>` is hidden and nothing in the menu is showing.

Comment (ahmadabdolsaheb) [reply src/components/Nav.js:51]: Here is the discussion if you would like to take a look: https://github.com/freeCodeCamp/freeCodeCamp/issues/41373
Ultimately we would need to make the navigation vertical on coderadio for consistency and then add the icons only when they are collapsed; For now, we could remove all icons for simplicity.

Comment (bbsmooth) [reply src/components/Nav.js:51]: Thanks for the link to the issue. To be honest, I didn't quite find a good explanation there but that could just be me. Regardless, the icon is not technically a WCAG requirement (but it is considered a best practice) so if you don't want it there then I'll remove it.

File: src/assets/external-link.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg aria-hidden="true" focusable="false" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path fill="currentColor" d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"></path></svg>
```
Comment (ShaunSHamilton) [inline src/assets/external-link.svg:1]: When does this get used/rendered? Is this browser-specific?

Comment (bbsmooth) [reply src/assets/external-link.svg:1]: I originally had this icon next to the links in the nav menu to denote that they open new windows but then removed them after I found out this isn't the style that FCC uses. I just forgot to remove the svg file.

File: src/css/App.css [hunk 3]
```diff
@@ -558,21 +562,14 @@ footer {
 }
```
Comment (ShaunSHamilton) [inline src/css/App.css:562]: To be a pain, would you mind editing the class above to not have the `overflow-y: hidden` - from what I can tell, this causes a horizontal scroll bar, and is never needed.

Comment (bbsmooth) [reply src/css/App.css:562]: I went even a step further and made the nav menu responsive down to about 200px by turning the FCC logo into just the icon at very narrow widths.

**Review by ShaunSHamilton — APPROVED** [review]
I am very happy with this 👍 

Thank you, Bruce!

**Comment by ahmadabdolsaheb** [comment]
Solid work there, Bruce. 👍

**Pull request merged**